### PR TITLE
Show QR code in fullscreen

### PR DESCRIPTION
**Pull request by przybciu**
Issue #1150 
please review
It works in the following way:
after clicking create button to create qr code, the mouse become a pointer cursor and the qr preview can be clicked to enter full screen mode. You can exit by clicking esc button or by clicking "back" on mobile phone.

---
Internal Tracking ID: [EXPOSUREAPP-14227](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14227)

### CODE REVIEW

**Review by Ein-Tim — CHANGES_REQUESTED** [review]
While the idea is good, it needs some more work. Viewing the QR code in full screen looks very off on my monitor. 

Bildschirmfoto 2022-10-24 um 22 03 28

**Comment by brianebeling** [comment]
Hey, thanks for your first contribution! I made some slight code style improvement, but for some reason I'm lacking permission to push she changes.

Perhaps because I checked out this PR and you are working on a fork. I left you a change request instead.

@Ein-Tim It seems to display correctly for me in Chrome on a 2560x1440 screen. What are your specs / browser?

File: src/assets/js/eventregistration.js [hunk 1]
```diff
@@ -187,6 +187,19 @@ document.getElementById('printCode').addEventListener('click', function (e) {
...
+    elem.requestFullscreen();
+  } else if (elem.mozRequestFullScreen) {
+    elem.mozRequestFullScreen();
+  } else if (elem.webkitRequestFullscreen) {
+    elem.webkitRequestFullscreen();
+  } else if (elem.msRequestFullscreen) {
+    elem.msRequestFullscreen();
+  }
+});
+
```
Comment (brianebeling) [inline src/assets/js/eventregistration.js:202]: Could be improved with: 

```js
  var requestMethod = 
    element.requestFullScreen ||
    element.webkitRequestFullscreen ||
    element.mozRequestFullScreen ||
    element.msRequestFullscreen;

    if( requestMethod ) {
      requestMethod.apply(element);
    }
```

tested locally, still works :)

**Comment by Ein-Tim** [comment]
@brianebeling 

I'm using Safari on macOS 12.6 with a 24" display.

**Comment by brianebeling** [comment]
> @brianebeling
> 
> I'm using Safari on macOS 12.6 with a 24" display.

Thanks! We will check if we can reproduce the issue

**Comment by przybciu** [comment]
> > @brianebeling
> > I'm using Safari on macOS 12.6 with a 24" display.
> 
> Thanks! We will check if we can reproduce the issue

I'm going to install safari and check that

**Comment by przybciu** [comment]
> > > @brianebeling
> > > I'm using Safari on macOS 12.6 with a 24" display.
> > 
> > 
> > Thanks! We will check if we can reproduce the issue
> 
> I'm going to install safari and check that

To be honest it is very difficult to reproduce it without mac. I tried safari in windows but it does not work the same :(

**Comment by brianebeling** [comment]
> > > > @brianebeling
> > > > I'm using Safari on macOS 12.6 with a 24" display.
> > > 
> > > 
> > > Thanks! We will check if we can reproduce the issue
> > 
> > 
> > I'm going to install safari and check that
> 
> To be honest it is very difficult to reproduce it without mac. I tried safari in windows but it does not work the same :(

Sadly Safari behaves very differently across operating systems. On MacOS it uses WebKit, while on Windows it might use an outdated version because it is discontinued. Testing Safari is best done on Safari, there is no substitute. However, there are services that grant you access to a Safari instance running on Mac. Its a slow-ish experience, but it works. 

The only tool I have experience with in that regard is Playwright. Please note that this is strictly not a recommendation by the open-source team, but my personal recommendation for a quick test without costs. 

I will check back with the team to see if we can reproduce this on a Mac. If we come to no conclusion through research or reproduction, I'd suggest we merge the PR. Its not a breaking change and I doubt a lot of people using this feature on a daily basis will use a Mac for that (e.g., public check in screens). We can still take care of the issue if it comes up and someone reports it as a real issue. Maybe we will get some additional information then.

**Comment by brifemu** [comment]
Hey @przybciu, thanks so much for your contribution. You have done a great job! I would like to contribute a couple of ideas to improve this new functionality:

- Implement a small indicative text under the generated QR with the following message: (EN) Click on the QR code for full screen - (DE) Klicken Sie auf den QR-Code für Vollbild
- In case you are unable to fix the Safari error on macOS, disable this option just for Safari on macOS and hide the text mentioned in the previous point.

**Comment by brianebeling** [comment]
@przybciu Hey, we want to merge this as quickly as possible to quickly get it out, even without the text improvement.  Unfortunately, we can't merge your fork, but we want to give you the opportunity to create a branch with your work in our repository - so you will be correctly credited - and then we will merge it.

**Comment by MikeMcC399** [comment]
@przybciu 

Please check out the GitHub help https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

"Allow edits from maintainers" should be enabled in your pull request.

**Comment by brianebeling** [comment]
Thank you very much for your PR @przybciu! I created a new PR and closed this one, because of the permission trouble. 
New PR is here: https://github.com/corona-warn-app/cwa-website/pull/3244

**Pull request closed without merging**